src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,380 @@
+    short_forecast: String,
+}
+
+impl ApiForecast {
+    fn wind_direction(&self) -> f64 {
+        let dir = match self.wind_direction.as_str() {
+            "NNE" => 1,
+            "NE" => 2,
+            "ENE" => 3,
+            "E" => 4,
+            "ESE" => 5,
+            "SE" => 6,
+            "SSE" => 7,
+            "S" => 8,
+            "SSW" => 9,
+            "SW" => 10,
+            "WSW" => 11,
+            "W" => 12,
+            "WNW" => 13,
+            "NW" => 14,
+            "NNW" => 15,
+            _ => 0,
+        };
+        (dir as f64) * (360.0 / 16.0)
+    }
+
+    fn icon_to_word(icon: WeatherIcon) -> String {
+        match icon {
+            WeatherIcon::Clear { .. } => "Clear",
+            WeatherIcon::Clouds { .. } => "Clouds",
+            WeatherIcon::Fog { .. } => "Fog",
+            WeatherIcon::Thunder { .. } => "Thunder",
+            WeatherIcon::Rain { .. } => "Rain",
+            WeatherIcon::Snow => "Snow",
+            WeatherIcon::Default => "Unknown",
+        }
+        .to_string()
+    }
+
+    fn wind_kmh(&self) -> f64 {
+        if self.wind_speed.unit_code.ends_with("km_h-1") {
+            self.wind_speed.value
+        } else {
+            self.wind_speed.value * MPH_TO_KPH
+        }
+    }
+
+    fn apparent_temp(&self) -> f64 {
+        let temp = if self.temperature.unit_code.ends_with("degC") {
+            self.temperature.value
+        } else {
+            (self.temperature.value - 32.0) * 5.0 / 9.0
+        };
+        let humidity = self.relative_humidity.value;
+        let wind_speed = self.wind_kmh();
+        australian_apparent_temp(temp, humidity, wind_speed)
+    }
+
+    fn to_moment(&self) -> WeatherMoment {
+        let icon = short_forecast_to_icon(&self.short_forecast, !self.is_daytime);
+        let weather = Self::icon_to_word(icon);
+        WeatherMoment {
+            icon,
+            weather,
+            weather_verbose: self.short_forecast.clone(),
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+
+    fn to_aggregate(&self) -> ForecastAggregate {
+        ForecastAggregate {
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+}
+
+fn combine_forecasts(data: &[ForecastAggregate], fin: WeatherMoment) -> Forecast {
+    let mut temp = 0.0;
+    let mut apparent = 0.0;
+    let mut humidity = 0.0;
+    let mut wind_north = 0.0;
+    let mut wind_east = 0.0;
+    let mut wind_kmh_north = 0.0;
+    let mut wind_kmh_east = 0.0;
+    let mut max = ForecastAggregate {
```

Comment:
I'd use `f64::MIN` for the `f64`s and `None` for the `wind_direction`

Reply:
(and `f64:MAX` for the min)

Reply:
Done, the minimums all default to `f64::MAX` and the maximums all default to `f64::MIN`. Both start with `None` for `wind_direction`.